File: src/Samples/Scheduler.ASP.NET.VB/Save.ashx.vb
Imports System.Web
Imports System.Web.Services
Imports DHTMLX.Common

Public Class Save
    Implements System.Web.IHttpHandler

    Sub ProcessRequest(ByVal context As HttpContext) Implements IHttpHandler.ProcessRequest

        context.Response.ContentType = "text/xml"
        Dim action = New DataAction(context.Request.Form)

        Dim data As New SchedulerDataContext()
        Try

            Dim changedEvent = DirectCast(DHXEventsHelper.Bind(GetType([Event]), context.Request.Form), [Event])
            Select Case action.Type
                Case DataActionTypes.Insert
                    data.Events.InsertOnSubmit(changedEvent)
                    Exit Select
                Case DataActionTypes.Delete
                    changedEvent = data.Events.SingleOrDefault(Function(ev) ev.id = action.SourceId)
                    data.Events.DeleteOnSubmit(changedEvent)
                    Exit Select
                Case Else
                    ' "update"                          
                    Dim eventToUpdate = data.Events.SingleOrDefault(Function(ev) ev.id = action.SourceId)
                    DHXEventsHelper.Update(eventToUpdate, changedEvent, New List(Of String)() From {"id"})
                    Exit Select
            End Select
            data.SubmitChanges()
            action.TargetId = changedEvent.id
        Catch a As Exception
            action.Type = DataActionTypes.[Error]
        End Try

        context.Response.Write(New AjaxSaveResponse(action).ToString())

    End Sub

    ReadOnly Property IsReusable() As Boolean Implements IHttpHandler.IsReusable
        Get
            Return False
        End Get
    End Property

End Class

File: src/Samples/Scheduler.ASP.NET/Save.ashx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using DHTMLX.Common;
using DHTMLX.Helpers;
using System.Globalization;

namespace SchedulerNetAsp
{
    /// <summary>
    /// </summary>
    public class Save : IHttpHandler
    {

        public void ProcessRequest(HttpContext context)
        {

           
            var action = new DataAction(context.Request.Form);
            var data = new SchedulerDataContext();

            try
            {

                var changedEvent = (Event)DHXEventsHelper.Bind(typeof(Event), context.Request.Form);//create event object from request

                switch (action.Type)
                {
                    case DataActionTypes.Insert: // define here your Insert logic
                        data.Events.InsertOnSubmit(changedEvent);
                        
                        break;
                    case DataActionTypes.Delete: // define here your Delete logic
                        changedEvent = data.Events.SingleOrDefault(ev => ev.id == action.SourceId);
                        data.Events.DeleteOnSubmit(changedEvent);
                        break;
                    default:// "update" // define here your Update logic
                        var updated = data.Events.SingleOrDefault(ev => ev.id == action.SourceId);
                        //update "updated" object by changedEvent's values, 'id' should remain unchanged
                        DHXEventsHelper.Update(updated, changedEvent, new List<string>() { "id" });
                        break;
                }
                data.SubmitChanges();
                action.TargetId = changedEvent.id;
            }
            catch
            {
                action.Type = DataActionTypes.Error;
            }

            context.Response.ContentType = "text/xml";
            context.Response.Write(new AjaxSaveResponse(action).ToString());
        }

        public bool IsReusable
        {
            get
            {
                return false;
            }
        }
    }
}